dataset/map_view.cpp

```diff
@@ -113,6 +113,31 @@ Dict<Key, Value> Dict<Key, Value>::slice(const Slice &params) const {
   return Dict(m_sizes.slice(params), slice_map(m_sizes, m_items, params));
 }
 
+template <class Key, class Value>
+Dict<Key, Value> &Dict<Key, Value>::setSlice(const Slice s, const Dict &dict) {
+  using core::to_string;
+  using units::to_string;
+  for (const auto &[key, item] : dict) {
+    const auto it = find(key);
+    if (it != end()) {
+      if (it->second.is_readonly() || !it->second.dims().contains(s.dim())) {
+        if (!it->second.dims().contains(s.dim()) || it->second.slice(s) != item)
+          throw except::DimensionError(
+              "Cannot update meta data '" + to_string(key) +
+              "' via slice since it is implicitly broadcast "
+              "along the slice dimension '" +
+              to_string(s.dim()) + "'.");
+      } else {
+        it->second.setSlice(s, item);
+      }
```

Comment:
I don't undersant your last paragraph. We can't overwrite the slice if it is readonly?

Reply:
Does `it->second.setSlice(s, item);` check again if it can be written? If so, why do it twice? If not, it overwrites the stuff in s with the same values, i.e. does nothing at the end of the day. My point is that this is avoided in your code but I am not sure how important it is.

Reply:
Yes it will check again. The reason for checking manually beforehand was to have a more specific error message.

Reply:
In that case, we should at least avoid the expensive check `it->second.slice(s) != item`, I think

Reply:
But then `setSlice` is called and fill throw an exception.

Reply:
Yes, but the code only checked once instead of twice.

Reply:
I am not sure what you are trying to say?